benchmark_runner/workloads/bootstorm_vm.py

```diff
@@ -25,36 +22,65 @@ def __init__(self):
         self.__data_dict = {}
+        self.__scale = int(self._scale)
 
-    def __create_vm_scale(self, vm_num: str):
+    def __wait_for_vm_start(self, vm_num: str):
         """
-        This method creates vms in parallel
+        This method wait for vm start in parallel
+        @return:
         """
-        self._oc._create_async(yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.__name}_{vm_num}.yaml'))
-        self._oc.wait_for_vm_status(vm_name=f'bootstorm-vm-{self._trunc_uuid}-{vm_num}', status=VMStatus.Stopped)
+        self._virtctl.wait_for_vm_status(vm_name=f'bootstorm-vm-{self._trunc_uuid}-{vm_num}', status=VMStatus.Running)
 
-    def __run_vm_scale(self, vm_num: str):
+    def __wait_for_vm_login(self, vm_num: str):
         """
-        This method runs vms in parallel
+        This method wait for vm login in parallel
+        @return:
         """
         vm_name = f'bootstorm-vm-{self._trunc_uuid}-{vm_num}'
-        self.set_bootstorm_vm_start_time(vm_name=vm_name)
-        self._virtctl.start_vm_sync(vm_name=vm_name)
         self.__data_dict = self.get_bootstorm_vm_elapse_time(vm_name=vm_name)
         self.__status = 'complete' if self.__data_dict else 'failed'
+        # upload to elasticsearch
         if self._es_host:
             self._upload_to_elasticsearch(index=self.__es_index, kind=self.__kind, status=self.__status, result=self.__data_dict)
             # verify that data upload to elastic search according to unique uuid
             self._verify_elasticsearch_data_uploaded(index=self.__es_index, uuid=self._uuid)
 
-    def __delete_vm_scale(self, vm_num: str):
+    def __create_vm_scale(self):
         """
-        This method deletes vms in parallel
+        This method creates vms
         """
-        vm_name = f'bootstorm-vm-{self._trunc_uuid}-{vm_num}'
-        self._oc.delete_vm_sync(
-            yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.__name}_{vm_num}.yaml'),
-            vm_name=vm_name)
+        # create vms
+        for vm_num in range(len(self._scale_node_list) * self.__scale):
+            self._oc._create_async(yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.__name}_{vm_num+1}.yaml'))
+            self._oc.wait_for_vm_status(vm_name=f'bootstorm-vm-{self._trunc_uuid}-{vm_num+1}', status=VMStatus.Stopped)
+
+    def __run_vm_scale(self):
+        """
+        This method runs vms, parallel only wait for vm start and wait for login
+        """
+        # start vms
+        for vm_num in range(len(self._scale_node_list) * self.__scale):
+            self.set_bootstorm_vm_start_time(vm_name=f'bootstorm-vm-{self._trunc_uuid}-{vm_num+1}')
+            self._virtctl.start_vm_async(vm_name=f'bootstorm-vm-{self._trunc_uuid}-{vm_num+1}')
+        # parallel the wait for vm start and login
+        proc = []
+        for target in [self.__wait_for_vm_start, self.__wait_for_vm_login]:
+            for vm_num in range(len(self._scale_node_list) * self.__scale):
```

Comment:
I don't see where this restricts how many VMs can be started at a time.

Reply:
Fixed